specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                            CHECK_FAILURE(args->get_InitiatingOrigin(&initiating_origin));
+                            std::wstring message = L"Launching External Protocol request";
+                            if (initiating_origin.get() == L"")
+                            {
+                                message += L"from ";
+                                message += initiating_origin.get();
+                            }
+                            message += L" to ";
+                            message += uri.get();
+                            message += L"?\n\n";
+                            message += L"Do you want to grant permission?\n";
+                            int response = MessageBox(
+                                nullptr, message.c_str(), L"Launching External Protocol",
+                                MB_YESNOCANCEL | MB_ICONWARNING);
+                            if (response == IDYES)
+                            {
+                                std::wstring protocol_url = uri.get();
+                                SHELLEXECUTEINFO info = {sizeof(info)};
+                                info.fMask = SEE_MASK_NOASYNC;
+                                info.lpVerb = L"open";
+                                info.lpFile = protocol_url.c_str();
+                                info.nShow = SW_SHOWNORMAL;
+                                ::ShellExecuteEx(&info);
+                            }
+                        }
+                        return S_OK;
+                    };
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    m_appWindow->RunAsync(
+                        [deferral, showDialog]()
+                        {
+                            showDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+}
+    
+```
+
+## .NET and WinRT
+
+```c#
+private WebView2 webView;
+void RegisterLaunchingExternalProtocolHandler() 
+{
+    webView.CoreWebView2.LaunchingExternalProtocol += (sender, args) {
+        {
+            CoreWebView2Deferral deferral = args.GetDeferral();
+            System.Threading.SynchronizationContext.Current.Post((_) =>
+            {
+                using (deferral)
+                {
+                    // Set the `Cancel` property to `TRUE`, as we will either silently launch the 
+                    // trusted app or display a custom dialog. 
+                    args.Cancel = true;
+                    if (String.Equals(args.Uri, "calculator:///", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If this matches our desired protocol, then set the
+                        // event args to cancel the event and launch the

```

Comment:
Remove "set the event args to cancel the event" since it's not done here any more.

Reply:
Please fix.